pallenec

```diff
@@ -7,28 +7,34 @@ local driver = require "pallene.driver"
 local print_ir = require "pallene.print_ir"
 local util = require "pallene.util"
 
+-- List of optimizations that the Pallene compiler does. If a new optimization is made,
+-- it should be added to this table.
+local all_opt_passes = {"constant_propagation"}
+
+-- List of compiler passes that are available. If a new optimization is made,
+-- it should be added to this table.
+local dump_passes = {"ast", "checker", "ir", "uninitialized", "constant_propagation"}
+
 local p = argparse("pallenec", "Pallene compiler")
 p:argument("source_file", "File to compile")
-p:flag("--emit-c", "Generate a .c file instead of an .so file")
-p:flag("--emit-asm", "Generate a .s file instead of an .so file")
-p:flag("--emit-lua", "Generate a .lua file instead of an .so file")
-p:flag("--compile-c", "Compile a .c file generated by --emit-c")
-p:flag("--show-locations", "Shows locations when dumping the AST")
-p:option("--dump", "Print the result of an internal compiler pass")
-    :choices({"ast", "checker", "ir", "uninitialized", "constant_propagation"})
-p:flag("--print-ir", "Show a human-readable version of the program's IR")
+p:mutex(
+    p:flag("--emit-c", "Generate a .c file instead of an .so file"),
+    p:flag("--emit-asm", "Generate a .s file instead of an .so file"),
+    p:flag("--emit-lua", "Generate a .lua file instead of an .so file"),
+    p:flag("--compile-c", "Compile a .c file generated by --emit-c"),
+    p:flag("--print-ir", "Show a human-readable version of the program's IR"),
+    p:option("--dump", "Print the result of an internal compiler pass"):choices(dump_passes)
+)
+p:option("--optimizations", "Pallene optimizations to be executed. If no optimization is listed, none is executed."):args("*")
+    :choices(all_opt_passes)
 local args = p:parse()
 
 -- For compilation errors that don't happen inside a source file.
 -- Inspired by gcc, eg. "gcc: fatal error: no input files".
 local compiler_name = arg[0]
 
-local function pallenec_abort(fmt, ...)
-    return util.abort(compiler_name .. ": " .. string.format(fmt, ...))
-end
-
-local function compile(in_ext, out_ext)
-    local ok, errs = driver.compile(compiler_name, in_ext, out_ext, args.source_file)
+local function compile(in_ext, out_ext, opt_passes)
```

Comment:
This function doesn't need to receive opt_passes as a parameter, because it is already in scope.